clang/lib/Driver/Driver.cpp

```diff
@@ -1189,18 +1190,19 @@ static bool addSYCLDefaultTriple(Compilation &C,
   llvm::Triple DefaultTriple =
       C.getDriver().getSYCLDeviceTriple(getDefaultSYCLArch(C));
   for (const auto &SYCLTriple : SYCLTriples) {
+    // Check current set of triples to see if the default has already been set.
     if (SYCLTriple == DefaultTriple)
       return false;
-    // If we encounter a known non-spir* target, do not add the default triple.
-    if (SYCLTriple.isNVPTX() || SYCLTriple.isAMDGCN())
-      return false;
-  }
-  // Check current set of triples to see if the default has already been set.
-  for (const auto &SYCLTriple : SYCLTriples) {
+    // TODO: Do we need this check? Don't we already detect the default triple
+    // in the previous condition?
     if (SYCLTriple.getSubArch() == llvm::Triple::NoSubArch &&
         SYCLTriple.isSPIROrSPIRV())
       return false;
+    // If we encounter a known non-spir* target, do not add the default triple.
+    if (SYCLTriple.isNVPTX() || SYCLTriple.isAMDGCN())
+      return false;
   }
+  // TODO: Why do we insert at the beginning? This is expensive operation.
```

Comment:
insert at the beginning isn't required.  Just need to be sure that the use knows where the triple is added (there is one more use in Driver.cpp).  It may also impact toolchain ordering which will require test updates.

Reply:
All tests pass unmodified. 😲

Reply:
We must not have instances where this is tested (i.e. multiple spir64_gen, spir64_x86_64 without spir64).